src/elementRoleMap.js

```diff
@@ -23,7 +23,25 @@ for (let i = 0; i < keys.length; i++) {
       if (relation.module === 'HTML') {
         const concept = relation.concept;
         if (concept) {
-          elementRoles.push([concept, [key]]);
+          const elementRoleRelation: ?ElementARIARoleRelationTuple = elementRoles.find(relation => ariaRoleRelationConceptAttributeEquals(relation[0], concept));
+          let roles: RoleSet;
+
+          if (elementRoleRelation) {
+            roles = elementRoleRelation[1];
+          } else {
+            roles = [];
+          }
+          let isUnique = true;
+          for (let i = 0; i < roles.length; i++) {
+            if (roles[i] === key) {
+              isUnique = false;
+              break;
+            }
+          }
+          if (isUnique) {
+            roles.push(key);
+          }
+          elementRoles.push([concept, roles]);
```

Comment:
Per my other comment, I think we could consider something like:

```suggestion
          if (!elementRoleRelation) {
            elementRoles.push([concept, roles]);
          }
```